src/nos/core/notifications.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""The content of all notification and confirmation emails."""
+
+import logging
+from typing import NamedTuple
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    "Notification",
+    "ACCESS_REQUEST_CREATED_TO_USER",
+    "ACCESS_REQUEST_CREATED_TO_DS",
+    "ACCESS_REQUEST_ALLOWED_TO_USER",
+    "ACCESS_REQUEST_ALLOWED_TO_DS",
+    "ACCESS_REQUEST_DENIED_TO_USER",
+    "ACCESS_REQUEST_DENIED_TO_DS",
+]
+
+
+class Notification(NamedTuple):
+    """A notification with a subject and a body text."""
+
+    subject: str
+    text: str
+
+    def format_text(self, **kwargs):
```

Comment:
Maybe this could be called "formatted" and return a Notification object with the text formatted. This formatted notification could then be directly passed to the emitter.notify method.

Reply:
Nice, see [2794cd1](https://github.com/ghga-de/notification-orchestration-service/pull/1/commits/2794cd1a0d7fc584c0c1ff1d393db867e79f15d5)

Reply:
Thanks, that's exactly what I had in mind.